src/main/java/com/redhat/cloud/policies/app/rest/PolicyCrudService.java

```diff
@@ -232,18 +237,36 @@ public Response getPoliciesForCustomer() {
     try {
       Pager pager = PagingUtils.extractPager(uriInfo);
       page = Policy.pagePoliciesForCustomer(entityManager, user.getAccount(), pager);
-      // TODO once the engine supports batching, rewrite this.
-      page.forEach(p -> {
-        try {
-          FullTrigger ft = engine.fetchTrigger(p.id, user.getAccount());
-          if (ft.conditions != null && !ft.conditions.isEmpty()) {
-            p.setLastEvaluation(ft.conditions.get(0).lastEvaluation);
+
+      // Get lastTriggeredTime and add it to the policies
+      String idsToFetch = page.stream().map(p -> p.id.toString()).collect(Collectors.joining(","));
+      try {
+        List<Trigger> list = engine.findTriggersById(idsToFetch,   user.getAccount());
```

Comment:
When the engine throws an error 500, this makes this request to fail too, not sure if that's expected or desired.

Reply:
I don't get the comment.
If the engine call in line 244 fails, then lastTriggered time will be 0.

Reply:
I had a bad/old engine running (fixed with a `mvn clean install`). All the requests failed here.


The `} catch (Exception e) {  p.setLastEvaluation(0);  }` clause is removed and the `ProcessingException|NotFoundException pe`  does not catch this error.